modules/tensor_mechanics/src/functions/InitialStrikeShearStress.C

```diff
@@ -0,0 +1,47 @@
+/*
+Define Function for Initial Shear Stress along Strike Direction
+Problem-Specific: TPV205-2D
+*/
+
+#include "InitialStrikeShearStress.h"
+
+registerMooseObject("TensorMechanicsApp", InitialStrikeShearStress);
+
+InputParameters
+InitialStrikeShearStress::validParams()
+{
+  InputParameters params = Function::validParams();
+  return params;
+}
+
+InitialStrikeShearStress::InitialStrikeShearStress(const InputParameters & parameters)
+  : Function(parameters)
+{
+}
+
+Real
+InitialStrikeShearStress::value(Real /*t*/, const Point & p) const
+{
+
+  Real x_coord = p(0);
+
+  double T1_o = 0.0;
+  if ((x_coord<=(0.0+1.5e3))&&(x_coord>=(0.0-1.5e3)))
+  {
+      T1_o = 81.6e6;
```

Comment:
these should probably be parameters, not hard coded

Reply:
I see. Here the function is problem specific and I don't expect this function to be general (since people may have different ways to setup the problem and nucleate the faults). Should I make it general or keep it as an example?

Reply:
The function has a lot of freedom depends on the problem, Here I just want to provide an example, but I would like to receive suggestions from reviewers